### PR TITLE
Population component refactor

### PR DESCRIPTION
**Pull request by albrja**
## Refactor Population Component

###\Makes Population a sub-class of Component
- *Category*: Refactor
- *JIRA issue*: [MIC-4414](https://jira.ihme.washington.edu/browse/MIC-4414)
- *Research reference*: 

### Changes and notes
-updates pins for vivarium/vph
-updates pins for pandas and numpy
-refactors population component to be a subclass of Component

### Verification and Testing
-Ran minimum simulation with population and household components. All tests pass

### CODE REVIEW

File: tests/test_synthetic_pii.py [hunk 1]
```diff
@@ -167,6 +167,7 @@ def test_first_and_middle_names(mocker, given_names, fake_obs_data):
     assert (first_names["first_name"] != middle_names["middle_name"]).any()
 
 
+@pytest.mark.slow
```
Comment (albrja) [inline tests/test_synthetic_pii.py:170]: The culprit.

File: tests/components/observers/test_household_survey_observers.py [hunk 2]
```diff
@@ -96,6 +96,8 @@ def test_get_observation(
     expected["middle_initial"] = ["J", "M"]
     expected[["guardian_1_address_id", "guardian_2_address_id"]] = np.nan
     expected["survey_date"] = [pd.to_datetime(sim_start_date)] * 2
+    # FIXME: Having dtype with with datime64[s] and [ns] causes pd.testing.assert_frame_equal to fail
+    expected["survey_date"] = expected["survey_date"].astype("datetime64[s]")
```
Comment (patricktnast) [inline tests/components/observers/test_household_survey_observers.py:100]: this is a nit, but is there a reason that setting the unit from to_datetime() wouldn't work?

**Review by patricktnast — APPROVED** [review]

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -15,7 +15,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.9", "3.10"]
```
Comment (stevebachmeier) [inline .github/workflows/build.yml:18]: Don't we typically include 3.8 as well?

File: tests/components/observers/test_household_survey_observers.py (hunk 2 above)
Comment (albrja) [reply tests/components/observers/test_household_survey_observers.py:100]: The string we are converting to datetime sim_start_date = "2021-01-01" doesn't have the seconds units and I tried a few different things and couldn't get it to work.

File: .github/workflows/build.yml (hunk 3 above)
Comment (albrja) [reply .github/workflows/build.yml:18]: We just need the latest 3 versions.

Comment (rmudambi) [reply .github/workflows/build.yml:18]: We usually support 3.8 as well, but there may have been some build issues with 3.8? I can't remember the specifics.

File: setup.py [hunk 4]
```diff
@@ -15,14 +15,14 @@
...
-        "vivarium_public_health==0.10.22",
+        "vivarium>=2.0.0",
+        "vivarium_public_health>=2.0.0",
         "click",
         "gbd_mapping>=3.0.6, <4.0.0",
         "jinja2",
         "loguru",
         "numpy",
-        "pandas>=1.0.0, <2.0.0",
+        "pandas",
```
Comment (rmudambi) [inline setup.py:25]: Unless you know the reason for the lower bound pin no longer applies, I'd suggest we leave it.

**Review by rmudambi — APPROVED** [review]

**Pull request merged**